x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -165,7 +166,8 @@ public final PhysicalOperation sourcePhysicalOperation(EsQueryExec esQueryExec,
         assert esQueryExec.estimatedRowSize() != null : "estimated row size not initialized";
         int rowEstimatedSize = esQueryExec.estimatedRowSize();
         int limit = esQueryExec.limit() != null ? (Integer) esQueryExec.limit().fold() : NO_LIMIT;
-        if (sorts != null && sorts.isEmpty() == false) {
+        boolean scoring = EsqlCapabilities.Cap.METADATA_SCORE.isEnabled() && esQueryExec.scoring();
```

Comment:
What would happen if the user requests the `_score` field, but `METADATA_SCORE` was not enabled? We'll probably want to rather not allow that (conditionally add to `MetadataAttribute.ATTRIBUTES_MAP`), and remove the check here.

Reply:
if the user requests _score without declaring the related metadata attribute, the query will fail with a verification exception like this: 
> Found 1 problem, line 2:91: Unknown column [_score]"""

Reply:
> without declaring the related metadata attribute

I meant that it would be a problematic execution if the user _does_ declare it, but `METADATA_SCORE` isn't enabled. I.e. on a non-snapshot built.

Reply:
the MetadataAttribute.ATTRIBUTES_MAP uses static initialization which makes it efficient, conditionally adding _score requires us to change how that is initialized (e.g., explicit init, something like `MetadataAttribute.ATTRIBUTES_MAP.getInstance()`, synchronized, most probably). also `EsqlCapabilities` sits outside of esql.core hence we can't use that in `EsPhysicalOperationProviders` without adding that (circular) dependency.
all in all, it seems to me that the tradeoffs suggest we let it like that and drop the `EsqlCapabilities` as soon as this is not usable in snapshots only anymore.

Reply:
thanks @ioanatia , I had lost it previously. committed your suggestion in [a0a1150](https://github.com/elastic/elasticsearch/pull/113120/commits/a0a1150753a82060831cb835ba51ef8e8e025305).